GalilSup/src/GalilController.cpp

```diff
@@ -3032,10 +2992,100 @@ bool GalilController::checkGalilThreads()
 			}
             return result;
 }
+
+/*--------------------------------------------------------------*/
+/* Find and replace text in string                              */
+/*--------------------------------------------------------------*/
+string GalilController::findReplace(string s, const string &toReplace, const string &replaceWith)
+{
+	while(s.find(toReplace) != std::string::npos) {
+		s = s.replace(s.find(toReplace), toReplace.length(), replaceWith);
+	}
+	return s;
+}
```

Comment:
replace() modifies the string, so s= is not needed. For better efficiency you could also pass string& to findReplace()

Reply:
hanks. Didn't realise replace modifies it in place.  I thought about passing in `string&` but I decided against it because I prefer not to mutate input arguments because it frequently obfuscates the side-effects of functions. The performance hit seemed marginal. Given the library version of `str::replace` does it though, it does make more sense to follow the same pattern.

Reply:
Maybe I'm old hat, but I don't like returning objects by value ... shared pointers are a good option when you have to.